### PR TITLE
Improve testing experience

### PR DESCRIPTION
**Pull request by chmeliik**
# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] New code has type annotations
- [ ] Docs updated (if applicable)
- [ ] Docs links in the code are still valid (if docs were updated)

### CODE REVIEW

**Comment by chmeliik** [comment]
If anyone else feels that running integration tests is very clunky right now, here's an idea

File: Makefile [hunk 1]
```diff
@@ -15,22 +15,28 @@ venv:
...
 test:
-	PATH="${PWD}/venv/bin:${PATH}" tox
+	venv/bin/tox
 
 test-unit:
-	PATH="${PWD}/venv/bin:${PATH}" tox -e $(TOX_ENVLIST) -- $(TOX_ARGS)
+	venv/bin/tox -e $(TOX_ENVLIST) -- $(TOX_ARGS)
+
+test-integration:
+	venv/bin/tox -e integration
```
Comment (brunoapimentel) [inline Makefile:25]: With this change here, the tests will always build and use a `cachi2:inttest` image, whereas the `make build-image` command will build a `cachi2:latest` image. Was that the intended effect? It feels to me that the build-image[-pristine] commands are not very useful now.

**Comment by brunoapimentel** [comment]
Big thanks for this, I was specially mad about having the image being completely rebuilt after every single change to fix an integration test.

File: Makefile (hunk 1 above)
Comment (brunoapimentel) [reply Makefile:25]: All in all, I like to have the image rebuilt (with cache) and the integration tests run with a single `make test-integration` command.

Comment (chmeliik) [reply Makefile:25]: Yeah, that's true. I felt like someone could be surprised if running integration tests rebuilt their `cachi2:latest` image, but looking back at it, I think that's reasonable behavior

Comment (chmeliik) [reply Makefile:25]: Updated to make int. tests build cachi2:latest by default

File: README.md [hunk 2]
```diff
@@ -214,23 +214,23 @@ In short, tox passes all arguments to the right of `--` directly to pytest.
 
 ### Running integration tests
 
-Build Cachi2 image and run all integration tests (but no other checks):
+Build Cachi2 image (locahost/cachi2:latest) and run all integration tests:
```
Comment (brunoapimentel) [inline README.md:217]: Typo: `locahost`

**Review by brunoapimentel — APPROVED** [review]

**Comment by chmeliik** [comment]
Rebased on main

**Comment by chmeliik** [comment]
@ejegrova found a bug where, if you already have `quay.io/containerbuildsystem/cachi2` pulled locally, then `podman build -t cachi2` doesn't tag the localhost image but the quay.io image :exploding_head: 

Fixed by passing localhost/cachi2 explicitly

**Review by ejegrova — APPROVED** [review]
LGTM :+1:

**Pull request merged**